src/main/java/com/google/sps/data/DBUtils.java

```diff
@@ -49,10 +61,35 @@ public static CollectionReference recipes() {
     return recipesReference;
   }
 
+  /** Return a list of all recipe IDs. */
+  public static ArrayList<String> allRecipeIds() {
+    ApiFuture<QuerySnapshot> querySnapshotFuture = recipesReference.get();
+    ArrayList<String> recipeIdList = new ArrayList<>();
+    QuerySnapshot querySnapshot = DBUtils.blockOnFuture(querySnapshotFuture);
+
+    if (querySnapshot == null) {
+      return null;
+    }
+
+    for (DocumentSnapshot document : querySnapshot.getDocuments()) {
+      recipeIdList.add(document.toObject(Recipe.class).id);
```

Comment:
We actually don't need to query any documents here, we can use `CollectionReference.listDocuments()` and `DocumentReference.getId()`